docs/trace/benchmark/Program.cs

```diff
@@ -14,147 +14,12 @@
-                .SetSampler(null)
-                .AddSource(this.sourceWithOneProcessor.Name)
-                .AddProcessor(new DummyActivityProcessor())
-                .Build();
-
-            Sdk.CreateTracerProviderBuilder()
-                .SetSampler(null)
-                .AddSource(this.sourceWithTwoProcessors.Name)
-                .AddProcessor(new DummyActivityProcessor())
-                .AddProcessor(new DummyActivityProcessor())
-                .Build();
-
-            Sdk.CreateTracerProviderBuilder()
-                .SetSampler(null)
-                .AddSource(this.sourceWithThreeProcessors.Name)
-                .AddProcessor(new DummyActivityProcessor())
-                .AddProcessor(new DummyActivityProcessor())
-                .AddProcessor(new DummyActivityProcessor())
-                .Build();
-        }
-
-        [Benchmark]
-        public void NoListener()
-        {
-            using (var activity = this.sourceWithNoListener.StartActivity("Benchmark"))
-            {
-                // this activity won't be created as there is no listener
-            }
-        }
-
-        [Benchmark]
-        public void PropagationDataListner()
-        {
-            using (var activity = this.sourceWithPropagationDataListner.StartActivity("Benchmark"))
-            {
-                // this activity will be created and feed into an ActivityListener that simply drops everything on the floor
-            }
-        }
-
-        [Benchmark]
-        public void AllDataListner()
-        {
-            using (var activity = this.sourceWithAllDataListner.StartActivity("Benchmark"))
-            {
-                // this activity will be created and feed into an ActivityListener that simply drops everything on the floor
-            }
-        }
-
-        [Benchmark]
-        public void AllDataAndRecordedListner()
-        {
-            using (var activity = this.sourceWithAllDataAndRecordedListner.StartActivity("Benchmark"))
-            {
-                // this activity will be created and feed into an ActivityListener that simply drops everything on the floor
-            }
-        }
-
-        [Benchmark]
-        public void OneProcessor()
-        {
-            using (var activity = this.sourceWithOneProcessor.StartActivity("Benchmark"))
-            {
-            }
-        }
-
-        [Benchmark]
-        public void TwoProcessors()
-        {
-            using (var activity = this.sourceWithTwoProcessors.StartActivity("Benchmark"))
-            {
-            }
-        }
-
-        [Benchmark]
-        public void ThreeProcessors()
-        {
-            using (var activity = this.sourceWithThreeProcessors.StartActivity("Benchmark"))
-            {
-            }
-        }
-    }
-
-    internal class DummyActivityProcessor : ActivityProcessor
+    public static void Main(string[] args)
     {
+        BenchmarkSwitcher.FromAssembly(typeof(Program).Assembly).Run(args);

```

Comment:
Changed this so you can run like:

```shell
dotnet run --configuration Release --filter SuppressInstrumentationBenchmarks
```

or 

```shell
dotnet run --configuration Release --filter TraceBenchmarks
```

Reply:
@alanwest FYI the benchmark project in /test/ is already set up for that. We should combine these two benchmark projects at some point.

Reply:
Yep noticed that, that's where I copied the pattern. Put these next to @reyang's other benchmarks for now since they are similar, but we can discuss a better place once we're through with analysis.

Reply:
Ping back from https://github.com/open-telemetry/opentelemetry-dotnet/pull/1039#issuecomment-671702276.